app/models/miq_queue.rb

```diff
@@ -366,7 +366,7 @@ def self.put_or_update(find_options)
     loop do
       msg = where_scope.order("priority, id").first
 
-      save_options = block_given? ? yield(msg, find_options) : nil
+      save_options = yield(msg, find_options) if block_given?
```

Comment:
Since save_options is in a loop, this effectively keeps the same value on each iteration of the loop, which is not what happened before where it was set to nil on every pass.  (I can't see if save_options is assigned elsewhere in this loop...if not, then this particular case is probably ok because the check is on `block_given?` which can't change during the loop.  I'm more pointing out that ternary `x = y ? z : nil` does not directly translate to the `x = z if y` and can have unexpected effects.)

Reply:
since this variable is declared in the loop, it will be reset.
But I prefer to keep it the old way

```ruby
>(1..9).each {|i| x = i if (i%2==0) ; puts x || "NIL" }
NIL
2
NIL
4
NIL
6
NIL
8
NIL
```

Reply:
reverted. this method is so touchy. don't want to mess with it.